src/elements/topographic-profile.ts

```diff
@@ -0,0 +1,138 @@
+import {LitElementI18n} from '../i18n.js';
+import {html} from 'lit';
+import {customElement, state} from 'lit/decorators.js';
+import i18next from 'i18next';
+import draggable from './draggable';
+import ToolboxStore from '../store/toolbox';
+import {cartesianToLv95} from '../projection';
+import * as d3 from 'd3';
+
+import type {GeometryAction} from '../store/toolbox';
+
+type ProfileServiceFormat = 'json' | 'csv'
+
+@customElement('topographic-profile')
+export class TopographicProfile extends LitElementI18n {
+  @state() hidden = true;
+  private coords: number[][] | undefined;
+  private name: string | undefined;
+
+  constructor() {
+    super();
+    super.hidden = this.hidden;
+    ToolboxStore.geometryAction.subscribe(options => this.handleActions(options));
+  }
+
+  connectedCallback() {
+    draggable(this, {
+      allowFrom: '.drag-handle'
+    });
+    super.connectedCallback();
+  }
+
+  update(changedProperties) {
+    if (changedProperties.has('hidden')) {
+      super.hidden = this.hidden;
+    }
+    super.update(changedProperties);
+  }
+
+  async handleActions(options: GeometryAction) {
+    if (options.action === 'profile') {
+      const geom = ToolboxStore.geometries.getValue().find(geom => geom.id === options.id);
+      this.coords = geom!.positions.map(c => cartesianToLv95(c));
+      this.name = geom!.name;
+      this.hidden = false;
+      await this.updateComplete;
+      this.plotProfile();
+    }
+  }
+
+  profileServiceUrl(format: ProfileServiceFormat) {
+    const url = new URL(`https://api3.geo.admin.ch/rest/services/profile.${format}`);
+    url.searchParams.set('geom', `{"type":"LineString","coordinates":[[${this.coords!.map(c => c.slice(0, 2).toString()).join('],[')}]]}`);
+    url.searchParams.set('sr', '2056');
+    url.searchParams.set('nb_points', '1000');
+    return url.toString();
+  }
+
+
+  plotProfile() {
+    // set the dimensions and margins of the graph
+    const margin = {top: 10, right: 10, bottom: 40, left: 60};
+    const width = 900 - margin.left - margin.right;
+    const height = 200 - margin.top - margin.bottom;
+
+    // append the svg object
+    const svg = d3.select('#profile-plot')
```

Comment:
You should externalize the d3 code to another file which would only depend on d3.

Reply:
Done in https://github.com/swissgeol/ngm/pull/752/commits/c98999c0583bd034d2635f40584bb7121841a94f